### ISSUE TITLE
Add option to remove light switch in HomeKit

### ISSUE THREAD

**Issue by Olivierbkk** [ratgdo/homekit-ratgdo#254]
Sorry to mix two topics...

I have trouble pairing with homekit, it keeps on saying the device is already paired. I reflashed the D1, unpaired in the web interface, nothing works. Any advice?

At some points I managed to have it working for a short time and noticed there was a light associated with the gate, can this be disabled, leaving only the gate in homekit? Note that I am planning to use it for a gate with dry contacts, not a garage door.

Thanks

**Comment by dkerr64** [comment]
For un-pairing / pairing I find that you need to first unpair (Remove accessory) from the Apple Home app.  It is not sufficient to just un-pair from the ratgdo.  So please try that.

Dry contact support is under development, you can track it in PR #253 and I'm sure we'll be looking for a tester soon!

Ability to disable the light switch is good idea.  Let's keep this issue open to track that... but I want to hold off adding this change until the dry contact support is merged.

Thanks

**Comment by Olivierbkk** [comment]
> For un-pairing / pairing I find that you need to first unpair (Remove accessory) from the Apple Home app. It is not sufficient to just un-pair from the ratgdo. So please try that.

I did that, it doesn’t show in apple home any more.

> Dry contact support is under development, you can track it in PR https://github.com/ratgdo/homekit-ratgdo/pull/253 and I'm sure we'll be looking for a tester soon!

I will be happy to test that for you, always happy to help! On a side note, an ESP32 version would be nice as you can get them with external wifi antenna, which helps considering that gates are usually a bit far from wifi access point (contrarily to garage doors). Not sure if ESP8266 boards have that option, I will check and order one if they exist. I understand that the ESP32 version is also under development, no worries, it is just my two cents.

> Ability to disable the light switch is good idea. Let's keep this issue open to track that... but I want to hold off adding this change until the dry contact support is merged.

Thanks! Not a priority obviously but would be nice.

**Comment by Olivierbkk** [comment]
> Not sure if ESP8266 boards have that option, I will check and order one if they exist.

They do exist, I just ordered one. Ready when you are...

**Comment by Olivierbkk** [comment]
btw once I know that this is working as I need I'll order the ratgdo board, for now I will just use an ugly breadboard.

**Comment by dkerr64** [comment]
Paul just launched the ESP32-based ratgdo board.  I have one and I have the HomeKit support almost ready (for equivalent function as the ESP8266-based board, new features not supported yet).  I will add dry contact to that once the pull request on the ESP8266 version is merged, I do not have the right garage door h/w to test it, so I am relying on the PR #253 author and any other testers we can find.

FYI that the ESP32 that Paul selected for his new ratgdo uses the built-in antenna and not one with external antenna connection.

Thanks.

**Comment by Olivierbkk** [comment]
I will be able to test with ESP8266 first and later on ESP32 if you want, no problem. I am just not good at coding though so it needs to be flashable directly to the board, either from the web as your current system or from arduino IDE. I will manage the hardware interface with the gate controller, don't worry about that, I'll just need to know which GPIO to use.
I believe there will be two inputs (gate open/closed switches) and two outputs (open and close commands). Let me know when you are ready.

**Comment by Olivierbkk** [comment]
I have received the new esp8266, I can test when you are ready (no pressure)

**Comment by dkerr64** [comment]
Dry contact support added to v1.8.4 and light switch is removed automatically in this mode (dry contact does not support light switch control).

v1.8.4 is available as a pre-release.  You will need to check the pre-release box on the firmware update dialog to see it.

**Comment by Olivierbkk** [comment]
Thanks, I have installed and selected pre release, was able to pair to Homekit.

I have looked around and can't find schematics or list of GPIO to connect to for the dry contact version, would you be able to point me to which GPIO is used for what?

**Comment by Olivierbkk** [comment]
OK so I found D5 and D6 for door open/close limit switches, this seems to be working. The door however always shows obstructed. I tried to connect D7, can't get rid of the "obstructed" status in homekit.

I yet to have to workout the outputs for opening and closing the gate (I guess there should be two output but can't find them in the code).

**Comment by dkerr64** [comment]
I think @tlhagan has not added obstruction detection to dry contact code.  I'll open a new issue to track that.

**Comment by PaulWieland** [comment]
Please share a picture of your wiring.

**Comment by Olivierbkk** [comment]
I am still testing on a breadboard, you can see here: https://github.com/ratgdo/homekit-ratgdo/issues/259#issuecomment-2543404864 

here are some pictures of the control board, the controls are 5VDC so I will probably use optocoupler or perhaps relays for the dry contact outputs, still need to test that.

What the control board offers is:
- O.P LIMIT: door open dry contact
- C.L LIMIT: door closed dry contact
These are inputs in the control board and will also be input in the ESP

- OP KEY: to start the opening motion
- CL KEY: to start the closing motion
- STP KEY: to stop opening/closing motion instantly
These are input in the control board and require only a short pulse signal

There is also an obstruction sensor connected to FSW, this sensor is a photo sensor that detects anything in the entrance, if there is detection while the door is closing, then the door will stop and open. This is probably not needed for the ratgdo to handle.

![IMG_6116 2](https://github.com/user-attachments/assets/32c98daa-d7ba-46fc-a956-d33c5ce64ab7)
![IMG_6124 2](https://github.com/user-attachments/assets/8bc50f1e-e54a-4db9-91d1-bddaf6e903a6)

**Comment by PaulWieland** [comment]
This is all way off topic then. This firmware is for ratgdo control boards.
